tempesta_fw/connection.h

```diff
@@ -77,22 +77,34 @@ enum {
  * @state	- connection processing state;
  * @list	- member in the list of connections with @peer;
  * @msg_queue	- queue of messages to be sent over the connection;
+ * @nip_queue	- queue of non-idempotent messages in server's @msg_queue;
  * @msg_qlock	- lock for accessing @msg_queue;
+ * @flags	- atomic flags related to server connection's state;
  * @refcnt	- number of users of the connection structure instance;
+ * @qsize	- current number of requests in server's @msg_queue;
  * @timer	- The keep-alive/retry timer for the connection;
  * @msg		- message that is currently being processed;
+ * @msg_sent	- message that was sent last in a server connection;
+ * @msg_resent	- message that was re-sent last in a server connection;
  * @peer	- TfwClient or TfwServer handler;
  * @sk		- an appropriate sock handler;
+ * @destructor	- called when a connection is destroyed;
+ * @forward	- called when a request is forwarded to server;
  */
-typedef struct {
+typedef struct tfw_connection_t {
 	SsProto			proto;
 	TfwGState		state;
 	struct list_head	list;
 	struct list_head	msg_queue;
+	struct list_head	nip_queue;				/*srv*/
 	spinlock_t		msg_qlock;
+	unsigned long		flags;					/*srv*/
 	atomic_t		refcnt;
+	atomic_t		qsize;					/*srv*/
```

Comment:
The reason this variable is made atomic is because it's checked in schedulers which run lock-less. The check is performed using `tfw_server_queue_full()` function in `server.h`. I will add a comment regarding this.